### ISSUE TITLE
Party affiliations in an Open Primary election

### ISSUE THREAD

**Issue by Josh-LACRRCC** [votinginfoproject/vip-specification#268]
California uses Open Primary voting for contests other than President / County Central Committees (aka Voter Nominated). As such, LA County differentiates between formal party nominated affiliation and candidates’ self-selected party preference.
( http://www.voterguide.sos.ca.gov/en/voter-info/elections-in-california.htm )
In order to help clarify to voters these differences, there is additional text as part of the Contest and Candidate party affiliation.

Using the California Governor as example:

In a traditional Partisan format:
  * Contest: Governor
  * Candidate: Jerry Brown – Democratic / DEM

And in a Voter Nominated format:
  * Contest: Governor – Voter Nominated
  * Candidate: Jerry Brown – Party Preference: Democratic / Pref: DEM
  <br/>or shortened<br/>
  * Candidate: Jerry Brown –Pref: Democratic / VN:DEM

The “Voter Nominated” identification text on the contest can easily be handled via the ContestBase.BallotSubTitle element. The additional text that accompanies the candidate affiliation raises a question of “how best to accomplish”. My first instinct is to create a copy of the root party, adjust the element content to fit, and then treat it like any other party using ID ref’s for its linkage to a candidate.

I.E.
```xml
  <Party id="par1">
    <Abbreviation>DEM</Abbreviation>
    <Name>
      <Text language="en">Democratic</Text>
    </Name>
  </Party>

  <Party id="par1vn">
    <Abbreviation>VN:DEM</Abbreviation>
    <Name>
      <Text language="en">Pref: Democratic</Text>
    </Name>
  </Party>
```

There is no master governing list of acceptable political parties. And active political parties can vary between jurisdictions. What are thought of as “Minor” political parties are treated no different, data wise. So, the Spec already supports variation in the list of parties.

Does this seem like an acceptable tactic? Other thoughts, concerns, alternatives?

**Comment by jdmgoogle** [comment]
This approach seems reasonable to me.

@jungshadow and @cjerdonek (since Chris will also have to deal with these)?

**Comment by jungshadow** [comment]
:+1: This sounds fine to me.

**Comment by cjerdonek** [comment]
Hmm, it sounds like the suggestion is to create a second version of each Party object: the original Party object and a VN-prefixed version of the party.

But the party is actually the same party in both cases.  It's just being used in a different way.  In one way, the party has nominated or chosen the candidate, but in the California Top-Two primary way, the candidate's party preference is just being listed.  But in both cases the party is the same.  Here is the text from the SOS web site:

> A candidate for nomination to a voter-nominated office shall have his or her party preference, or lack of party preference, stated on the ballot, but the party preference designation is selected solely by the candidate and is shown for the information of the voters only. It does not mean the candidate is nominated or endorsed by the party designated, or that there is an affiliation between the party and candidate, and no candidate nominated by the voters shall be deemed to be the officially nominated candidate of any political party.

Instead of creating a copy of each party, how about adding another property to the "CandidateSelection" object for the candidate's party preference as opposed to the party's preference?  It can be called something like "CandidatePartyId" or "CandidatePreferencePartyId":

```xml
<xs:complexType name="CandidateSelection">
  <xs:complexContent>
    <xs:extension base="BallotSelectionBase">
      <xs:sequence>
        <xs:element name="CandidateId" type="xs:IDREF" maxOccurs="unbounded" />
        <xs:element name="CandidatePartyId" type="xs:IDREF" minOccurs="0" />
        <xs:element name="EndorsementPartyId" type="xs:IDREF" minOccurs="0" maxOccurs="unbounded" />
        <xs:element name="IsWriteIn" type="xs:boolean" minOccurs="0" />
      </xs:sequence>
    </xs:extension>
  </xs:complexContent>
</xs:complexType>
```

**Comment by jdmgoogle** [comment]
> Instead of creating a copy of each party, how about adding another property to the "CandidateSelection" object for the candidate's party preference as opposed to the party's preference?

There is a `PartyId` field for the `Candidate` object. Would that work?

**Comment by jungshadow** [comment]
@cjerdonek What say you, sir? Is `PartyId` on `Candidate` good for the time being?

**Comment by cjerdonek** [comment]
> Is PartyId on Candidate good for the time being?

Yes, I think so. There is the question of whether to display that information beneath a candidate's name in a contest, but I think that nuance can wait.  I'm not sure the extent to which VIP is aiming to capture ballot display, etc.

**Comment by cjerdonek** [comment]
By the way, this discussion brings to mind a couple things that might be worth clarifying in the docs:

1. how to encode the lack of a party preference, as opposed to a preference that is simply missing (e.g. should jurisdictions encode ["no party preference"](http://www.sos.ca.gov/elections/political-parties/no-party-preference/) as a party with its own party ID)
2. is each candidate object supposed to appear in only than contest, or should a new candidate object be created for each contest?  I'm not sure if there is a way to represent such a constraint in the XSD. I'm also not sure how common this situation is.  (Recalls where the candidate is also up for election as the replacement is one possibility.)

**Comment by jdmgoogle** [comment]
> I'm not sure the extent to which VIP is aiming to capture ballot display, etc. 

IMHO the goal should be to capture enough information such that

1. Someone **not** familiar with the specific ballot layout details (e.g., "each candidate much have name listed, then their party, then profession, then endorsing party") should be able to create a reasonable facsimile of a generic-looking ballot; and
2. Someone who **is** familiar with specific ballot layout details should be able to recreate a ballot because all the necessary information is there.

In short, think of the content/presentation separation in web design. The content of the feed should contain all the relevant information, whereas the presentation of the information is left up to the web service. Google search will provide one "CSS", the Pew embeddable tool provides another "CSS", and if a jurisdiction wants to use information from the API (or process the raw XML) they can layer another CSS on top of that.

@Josh-LACRRCC and @cjerdonek: What else needs to be done/answered before we close this out?

**Comment by cjerdonek** [comment]
If the goal is to let people create a reasonable facsimile, then it seems like we'd need to put a flag on the CandidateContest object saying whether the following is true (i.e. whether candidate party preference should be displayed):

> A candidate for nomination to a voter-nominated office shall have his or her party preference, or lack of party preference, stated on the ballot, but the party preference designation is selected solely by the candidate and is shown for the information of the voters only.

(This is the language from the SOS link that @Josh-LACRRCC first posted above.)

Maybe this can be for a future VIP release since it is an addition to the spec.

**Comment by jdmgoogle** [comment]
I would argue that that's presentation logic, as opposed to content logic.

**Comment by cjerdonek** [comment]
Okay, that's fine with me.

**Comment by Josh-LACRRCC** [comment]
So, CandidateSelection.EndorsementPartyId would be the direct party endorsement (i.e. DNC/GOP endorsement for President) and, Canidate.PartyId would be the non-endorsed, candidate selected affiliation. The elements are optional, so that is easy enough to implement with while building a Feed file.

But, LAC see's the additional "party preference" text as a required piece of information. If a candidate's party info, in a Voter Nominated contest, is displayed, it must contain that denotation. I would not want to leave the logic of when to display this extra bit of text up to a "CSS". 
The use of duplicate parties with reformatted text ensures that the content is present in the file, doesn't require any changes to the Spec, and has minimal impact on total file size. It is a way to accomplish LA specific usage requirements without requiring others to conform to, or even be aware of, our standards.

**Comment by jdmgoogle** [comment]
@Josh-LACRRCC  I think ultimately we're in agreement this is going to be an issue of how whatever piece of code is actually presenting the data contained in the feed handles this situation, as well as how the feed itself is generated. If LAC has a way to generate feeds in a way that handles Voter Nominated contests correctly, then I think we're set here.

**Comment by cjerdonek** [comment]
I would rather see the spec changed than put counties in the situation of using the spec in the way that @Josh-LACRRCC is proposing.

For one, if LAC were to go that route, then LAC's list of parties would be different from every other county.  How would feed consumers know to skip over the "VN-prefixed" ones when displaying the list of parties to voters?

Also, someone using LAC's feed to auto-generate a sample ballot would unknowingly generate confusing (and incorrect) displays like--

> Candidate Name: Jerry Brown
Endorsed by party: VN-Dem

or perhaps--

> Candidate Name: Jerry Brown
Party preference: Dem
Endorsed by party: VN-Dem

The issue is that "EndorsementPartyId" is supposed to be filled in by the party endorsing the candidate (according to the spec), which wouldn't be the case here.

These are state-wide contests, so I think the way to handle it should be clear rather than each county coming up with an ad hoc approach.

**Comment by jdmgoogle** [comment]
Question: Does California have "Endorsements" the same way that New York does? Or is "voter-nominated" the way California does New-York-style endorsements?

**Comment by cjerdonek** [comment]
> Question: Does California have "Endorsements" the same way that New York does? Or is "voter-nominated" the way California does New-York-style endorsements?

I'm not sure what New York-style "endorsements" are, but no, I don't believe that California has them.  It's conceivable that some local elections in California have such a concept, though (e.g. partisan local elections).

Before 2012, California had partisan primaries for state offices, so some or all of the candidates on the ballot in the November general election were "party-nominated," which I guess would qualify as meaning "endorsed" by the party in this context.

However, in 2012, California starting using a "top two" primary where the top two candidates advance regardless of party.  So in the November general election, these candidates are now labeled "voter nominated" (as opposed to "party nominated"), since they are not the party's selection.  However, the candidates can still list the party they are registered under.

**Comment by jdmgoogle** [comment]
My preference -- since we're down to the wire -- is to try the 2015 cycle with the current schema and re-evaluate once a few different jurisdictions have a chance to play with real data (or data from 2014). This seems very California-specific and I'm incredibly hesitant to modify the schema by adding yet another place to list `PartyId` (it's already in `Person`, `Candidate`, and `CandidateSelection`).

Unless there's an incredibly compelling reason to not wait 3 months, let's wait 3 months.

**Comment by cjerdonek** [comment]
I don't see a rush either.  However, for the record, this type of primary system is also used in Louisiana and Washington states (and perhaps other states), in addition to California.  It's commonly called the "Top Two Primary."  See [here](https://en.wikipedia.org/wiki/Nonpartisan_blanket_primary) for more info.

As for modifying the schema, I wouldn't recommend adding another place for the party ID since the correct party ID already has a place under the candidate.  It's just a matter of flagging for which contests the candidate's party should be displayed (or as @Josh-LACRRCC requested, for which contests this information is mandatory for feed publishers).

**Comment by jdmgoogle** [comment]
Okay. In that case I'm going to bump this out to 5.1 and we can figure it out then.

**Comment by jdmgoogle** [comment]
@Josh-LACRRCC Provided an LA County data feed for us using 2014 data. Are you happy with how the spec itself was able to handle this data? Do you have any recommendations for best practices?

**Comment by jdmgoogle** [comment]
Ping @Josh-LACRRCC and @cjerdonek to pick up conversation again.

**Comment by jdmgoogle** [comment]
Friendly ping to @Josh-LACRRCC and @cjerdonek

**Comment by cjerdonek** [comment]
Going back to @Josh-LACRRCC's original request:

> LA County differentiates between formal party nominated affiliation and candidates’ self-selected party preference.

Is the spec clear enough as to whether `Candidate.PartyId` represents the "formal party nominated affiliation" (if it exists) or the "candidate's self-selected party preference"?  Or should the latter instead be drawn from `Candidate.Person.PartyId` (which I assume is the party as the person is registered)?  Either way, we should probably clarify in the docs.

@Josh-LACRRCC, in California, must the party preference a candidate lists in a voter-nominated contest always match the party they're registered in?

Some possible options, depending on the answers to the above questions:

1) Add an attribute to `Contest` saying how `Candidate.PartyId` should be interpreted (e.g. party-nominated affiliation vs. candidate's self-selected preference),
2) Add an attribute to `Contest` saying whether the candidate's `Person.PartyId` should be displayed, or
3) Do nothing, if the candidate's `Person.PartyId` is always displayed anyways (e.g. if mimicking the display of the ballot is out of scope for the spec).

**Comment by jdmgoogle** [comment]
Ping to @Josh-LACRRCC and @cjerdonek: I'm looking to wrap up 5.1 by the end of this week (i.e., tomorrow).

My personal preference is to treat the various party tags thusly:
1. `Person.PartyId` is how that person self-identifies. E.g., Bernie Sanders would be an "Independent".
2. `Candidate.PartyId` is the party under whose banner they're running. E.g., Sanders would be running as a "Democrat".
3. `CandidateSelection.EndorsementPartyId` would be how that candidate is presented on the ballot, if different than `Candidate.PartyId`. E.g., in New York or California this could be a cross-endorsement or the "Voter-Nominated" tag.

Does that make sense?

**Comment by Josh-LACRRCC** [comment]
A candidate is not required to be registered with the party they are self-identifying as in a "voter-nominated" contest.

Earlier in this thread, we gravitated towards `Candidate.PartyId` being self-identifying (voter nominated) and `CandidateSelection.EndorsementPartyId` being under the official party banner. The bigger issue is making sure accurate information is presented. "Voter Nominated" vs "Party Endorsement" is a crucial distinction that LAC wants to ensure is properly conveyed. 
This concern is mitigated by using the `CandidateContest.BallotSubTitle` to label a contest as "Voter Nominated". As long as the documentation clearly informs a feed consumer about the differences between the various party elements, the desired information can work within the existing schema.

In this discussion, there are 3 types of party implementations:
- Registered Party
- Self-identifying Party (voter nominated)
- Endorsed Party (given the "thumbs up" by an official political party)

There are also 3 solid elements for Party Ref's:
- Person.PartyId
- Candidate.PartyId
- CandidateSelection.EndorsementPartyId

We just need to agree to a mapping, and get it documented for the consumers.

**Comment by cjerdonek** [comment]
I would go with the following (which it sounds like is what @Josh-LACRRCC is suggesting):

* Registered Party: Person.PartyId
* Self-identifying Party (e.g. voter nominated): Candidate.PartyId
* Endorsed Party (given the "thumbs up" by an official political party):  CandidateSelection.EndorsementPartyId

**Comment by kennethmbennett** [comment]
In California, the only time the Candidate.PartyID would be different from Person.PartyID is when a candidate running for a Voter-Nominated office decides not to reveal their party affiliation.  In that case, the Party Affiliation would be presented as "None."  For example, under Voter-Nominated offices, it is not possible for a candidate to affiliate with a party to which they are not registered.  For the Presidential Primary (the only true primary left on the California ballot), the only party that matters is the Candidate.PartyID, at least as far as the content of the official ballot is concerned.  I have troubles understanding under what circumstances and how and election official anywhere would want to highlight a distinction between Person.PartyID and Candidate.PartyID, especially for official election materials.


With respect to endorsements, in California a party's endorsement of a candidate is different from the candidate's affiliation with a party.  The latter is the party with which the candidate is registered, while the former is a party's expression of support for a candidate.  But this is a one-to-many relationship.  Candidates can and often do get endorsed by multiple parties.


Candidates in voter-nominated contests can receive party endorsements, so not sure how that reconciles with using the CandidateSelection.EndorsementPartyID to store the "Voter-Nominated" tag.  The term voter-nominated is really an attribute of a contest or office, and not a candidate.  More specifically, it describes the primary nominating process for the office/contest (nominated by vote of all voters vs. nominated by vote of party members).



________________________________

**Comment by drrwebber** [comment]
I believe this requirement mirrors what OASIS EML is doing - providing ability to assign both. In EML this is there because in elections you can have parties (aka political entities) affiliating (sponsoring) candidates.  True this may not show up in the final ballot or results specifically - but is important in areas of funding, support, and candidate canvassing materials.

**Comment by jdmgoogle** [comment]
It appears that since no spec changes are necessary to solve this issue, I'm moving this over to the "5.1 Documentation" milestone.

@Josh-LACRRCC @kennethmbennett @cjerdonek I emailed the three of you just now to try and arrange a teleconference by or on this Wednesday (the 3rd) so we can sort this out. If no time is good for the three of you by then, I'll do my best to summarize this discussion and update the documentation for a 5.1 release by the end of the week.

**Comment by jdmgoogle** [comment]
**Update**

After discussing this with @Josh-LACRRCC and @cjerdonek here's the approach upon which we've settled:

These "top-two" or "non-partisan" primaries are a feature of the office or contest, not necessarily of the candidate themselves. In other words, there are partisans running in these contests, but the contest itself is open to people from all parties. Therefore:

1. `CandidateContest.PrimaryPartyId` will be used to specify that this contest is "Voter Nominated", "Nonpartisan", or whatever the terminology is in your state. The feed creator will create a `Party` element indicating that the contest is non-partisan, and associate the `CandidateContest` with that party. For example, California would create a `Party` with the name "Voter Nominated", and associate all "Top-Two" contests with that `PartyId`.
2. `Person.PartyId` will only be used to help data consumers disambiguate between different people. It will not be considered part of ballot data at all.
3. `Candidate.PartyId` will be the candidate's official party association, and is what would be shown on the ballot next to the candidate.
4. `CandidateSelection.EndorsingPartyIds` is used mostly (only?) in New York where political parties can endorse candidates from other parties, and have votes cast for that (candidate, party) combination count towards that party's "minimum viable support" threshold to be considered an official state-level party.

I'll update the sample feed to include an example of this.

**Comment by jdmgoogle** [comment]
Closed by #308.
